src/Workspaces/Core/Portable/SymbolKey/SymbolKey.ErrorTypeSymbolKey.cs

```diff
@@ -32,42 +29,53 @@ public static SymbolKeyResolution Resolve(SymbolKeyReader reader)
                 var name = reader.ReadString();
                 var containingSymbolResolution = reader.ReadSymbolKey();
                 var arity = reader.ReadInteger();
-                var typeArgumentResolutions = reader.ReadSymbolKeyArray();
 
-                var errorTypes = ResolveErrorTypes(reader, containingSymbolResolution, name, arity);
+                using var typeArguments = reader.ReadSymbolArray<ITypeSymbol>();
+                using var errorTypes = ResolveErrorTypes(reader, containingSymbolResolution, name, arity);
 
-                if (typeArgumentResolutions.IsDefault)
+                if (typeArguments.Count != arity)
+                {
+                    return default;
+                }
+
+                if (arity == 0)
                 {
                     return CreateSymbolInfo(errorTypes);
                 }
 
-                var typeArguments = typeArgumentResolutions.Select(
-                    r => GetFirstSymbol<ITypeSymbol>(r)).ToArray();
-                if (typeArguments.Any(s_typeIsNull))
+                using var result = PooledArrayBuilder<INamedTypeSymbol>.GetInstance();
+                var typeArgumentsArray = typeArguments.Builder.ToArray();
```

Comment:
unfortunate allocation.  However, the compiler has no way to call into their APis without doing this.